lntest/itest/lnd_test.go

```diff
@@ -9031,30 +9031,41 @@ out:
 	if err := net.ConnectNodes(ctxt, net.Bob, carol); err != nil {
 		t.Fatalf("unable to connect bob to carol: %v", err)
 	}
-	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
-	chanPoint = openChannelAndAssert(
-		ctxt, t, net, net.Bob, carol,
-		lntest.OpenChannelParams{
-			Amt: chanAmt,
-		},
-	)
 
 	// Reconnect Alice and Bob. This should result in the nodes syncing up
 	// their respective graph state, with the new addition being the
 	// existence of Carol in the graph, and also the channel between Bob
 	// and Carol. Note that we will also receive a node announcement from
 	// Bob, since a node will update its node announcement after a new
-	// channel is opened.
+	// channel is opened. We connect Alice and Bob here to ensure her
+	// gossip filter isn't set after the announcements between Bob and Carol
+	// are generated.
 	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
 	if err := net.EnsureConnected(ctxt, net.Alice, net.Bob); err != nil {
 		t.Fatalf("unable to connect alice to bob: %v", err)
 	}
 
+	// Alice shouldn't receive any new updates yet since the channel has yet
+	// to be opened.
+	select {
+	case <-graphSub.updateChan:
+		t.Fatalf("received unexpected update")
+	case <-time.After(time.Second):
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, channelOpenTimeout)
+	chanPoint = openChannelAndAssert(
+		ctxt, t, net, net.Bob, carol,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+		},
+	)
+
 	// We should receive an update advertising the newly connected node,
 	// Bob's new node announcement, and the channel between Bob and Carol.
 	numNodeAnns = 0
 	numChannelUpds = 0
-	for numChannelUpds < 2 && numNodeAnns < 1 {
+	for numChannelUpds < 2 || numNodeAnns < 2 {
```

Comment:
👍 

If you do this (and the other loop condition change) without moving the `openChannel` call, are tests still passing?

Reply:
It passes most of the time, but still flakes